### DISCUSSION TITLE
Issues while creating custom kamelet

### DISCUSSION THREAD

**Discussion by akrutikhandelwal** [apache/camel-karavan#1110] [General]
We are trying to create a custom kamelet of sending attachments through WhatsApp and also sending emails  where the body is based on an html file present. But we are unable to do this because of limitations while creating karavan visually and even with the yaml file. Is there any reference doc on how to build our yaml and what all properties are acceptable. Could find one doc of the schema (https://gist.github.com/lburgazzoli/4a74731d494ef6060e557402697b89cd but it is not ) but there are a lot of functionalities missing..

**Comment by davsclaus** [comment]
You can look at all the existing kamelets from the camel-kamelets and see how they are structured and build.
https://github.com/apache/camel-kamelets/tree/main/kamelets

Also it would help if you can be more specific what functionality is missing and what problems you have? 

Kamelets was born in Camel K so there is extensive docs at
https://camel.apache.org/camel-k/2.2.x/kamelets/kamelets.html

**Reply by akrutikhandelwal** [reply]
We hit a roadblock trying to send an email with an attachment using the SMTP Kamelet. Unfortunately, it lacked the functionality to directly attach files to the message body. No matter what we tried, we ended up with an email containing another email, and the attachment inside was garbled and unusable.

**Reply by davsclaus** [reply]
This is not a Karavan issue. This is a general camel + kamelets. You should ask for help there.

**Reply by akrutikhandelwal** [reply]
but even with Karavan there was no such option. The karavan integration generates the <name>.kamelet.yaml file and even with the yaml dsl we were not able to figure out a way to add an attachment

**Comment by davsclaus** [comment]
this is more of a general camel questions how to send attachments in emails
